core/includes/mail.inc

```diff
@@ -155,6 +155,34 @@ function backdrop_mail($module, $key, $to, $language, $params = array(), $reply
   }
   $message['headers'] = $headers;
 
+  // If requested set Reply-To to the From and re-format From address.
+  if (variable_get('site_mail_replyto') &&
+      empty($message['headers']['Reply-To']) &&
+      ($default_from_name = variable_get('site_name', 'Backdrop CMS')) &&
+      ($default_from_parts = explode('@', $default_from)) &&
+      count($default_from_parts) == 2 &&
+      $default_from_parts[1] &&
+      stripos($message['from'], '@' . $default_from_parts[1]) === FALSE) {
```

Comment:
This is one of the ugliest IF statements I've seen in a while. Let's add some nesting if needed and avoid the inline variable assignments.